tests/test_database.py

```diff
@@ -80,3 +79,33 @@ def test_current_timestamp(self):
         db = get_conn(self.db_cls)
         res = db.query(current_timestamp(), datetime)
         assert isinstance(res, datetime), (res, type(res))
+
+    def test_correct_timezone(self):
+        name = "tbl_" + random_table_suffix()
+        db = get_conn(self.db_cls)
+        tbl = table(db.parse_table_name(name), schema={
+            "id": int, "created_at": "timestamp_tz(9)", "updated_at": "timestamp_tz(9)"
+        })
+
+        db.query(tbl.create())
+
+        tz = pytz.timezone('Europe/Berlin')
+
+        now = datetime.now(tz)
+        db.query(table(db.parse_table_name(name)).insert_row("1", now, now))
+        db.query(db.dialect.set_timezone_to_utc())
+
+        t = db.table(name).query_schema()
```

Comment:
This is refreshing the schema, to actually get the correct timezones/datatype values.